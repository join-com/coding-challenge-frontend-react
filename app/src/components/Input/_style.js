import styled from 'styled-components';

export default styled.input`
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
`;
