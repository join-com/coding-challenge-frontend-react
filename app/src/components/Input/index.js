import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import StyledComponent from './_style';

const dateHandlers = {
    type: 'text',
    onFocus: e => e.target.type = 'date',
    onBlur: e => e.target.type = 'text'
};

export default function Input({
    className,
    classNameMod,
    ...rest
}) {
    const componentClassName = classnames('c-input', className, {
        [`c-input--${classNameMod}`]: classNameMod
    });

    return (
        <StyledComponent
            className={componentClassName}
            {...rest}
            {...rest.type === 'date'? dateHandlers : {}}
        />
    );
}

Input.propTypes = {
    className: PropTypes.string
};
