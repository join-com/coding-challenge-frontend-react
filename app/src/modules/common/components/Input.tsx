import * as React from 'react'

export type InputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  onValueChange?: (newValue: string) => void
}

export const Input: React.FC<InputProps> = ({ onValueChange, ...props }) => {
  function onChange(event: React.ChangeEvent<HTMLInputElement>) {
    props.onChange && props.onChange(event)
    onValueChange && onValueChange(event.target.value)
  }
  return (
    <>
      <style jsx={true}>
        {/* language=SCSS */ `
          input {
            border-radius: 0;
            background-color: #fff;
            border: 1px solid #ccc;
            padding: 8px;
            outline: none;
            width: 100%;
            display: inline-flex;
            font-size: 14px;
            height: 36px;
            &:focus {
              border-color: #0577ff;
            }
          }
        `}
      </style>
      <input onChange={onChange} {...props} />
    </>
  )
}
