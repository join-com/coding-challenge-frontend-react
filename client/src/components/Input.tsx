import React, { FC, useState } from "react";
import Styled from "styled-components";
interface IInput {
  className?: string;
  placeholder?: string;
  type?: string;
  postfix ?: any;
}

const Input: FC<IInput> = ({ className, type, postfix, placeholder }) => {
const [focus, setFocus] = useState(false);
const [content, setContent] = useState("");
return <div className={`${className} ${focus ? "focused" : ""} ${content.length ? "full" : "empty"}`}>
  <label>{placeholder}</label>
  <input type={type}
         onChange={(e) => setContent(e.target.value)}
         onFocus={() => setFocus(true)}
         onBlur={() => setFocus(false)}/>
  <span className="postfix">{postfix}</span>
  <div className="underline"/>
</div>;
};

export default Styled(Input)`
  padding: 5px;
  background: #FAFAFA;
  font-size: 1.1em;
  display: inline-block;
  height: 25px;
  margin: 10px;
  border-radius: 2px;
  .underline {
    width: 0%;
    transition: width 0.314s;
    border-bottom: 1px solid black;
  }
  label {
    width: auto;
    position: absolute;
    height: 20px;
    overflow: hidden;
    font-size: 100%;
    display: block;
    transition: font-size 0.314s;
    pointer-events: none;
    display: flex;
    flex-direction: column;
  }
  input {
    border: 0;
    background: transparent;
    height: 90%;
    &:focus{
      outline: none;
    }
    &[type="date"] {
      color: transparent;
    }
  }
  &.focused label, &.full label{
      font-size: 8px;
      flex-direction: row;
  }
  &.focused, &.full {
    input {
      &[type="date"] {
        color: black;
      }
    }
  }
  input:focus + span + div.underline {
    width: 100%;
  }
  .postfix {

  }
`;
