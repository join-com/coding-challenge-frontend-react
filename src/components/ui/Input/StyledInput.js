import styled from "styled-components";
import { defaultInputStyles, hoverBorderStyles } from "../../../helpers/stylesMixins";

export const StyledInput = styled.input`
  ${defaultInputStyles};
  width: 100%;
  box-sizing: border-box;
  &:focus {
    ${({ disabled }) => !disabled && hoverBorderStyles};
  }
`;
