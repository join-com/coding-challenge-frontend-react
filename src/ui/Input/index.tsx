import React, { FunctionComponent } from 'react'
// @ts-ignore
import { FieldTextStateless } from '@atlaskit/field-text'
import { FieldRenderProps } from 'react-final-form'

type FFProps = FunctionComponent<FieldRenderProps<any>>
type OwnProps = { input: object; placeholder: string }

type Props = FFProps & OwnProps

export const FFInput = ({ input, placeholder }: Props) => (
  <FieldTextStateless
    {...input}
    placeholder={placeholder}
    isLabelHidden
    shouldFitContainer
  />
)
