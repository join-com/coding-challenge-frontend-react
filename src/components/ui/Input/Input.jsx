import React from "react";
import PropTypes from "prop-types";
import { StyledInput } from "./StyledInput";

const Input = ({ disabled, placeholder, value, onChange, type, name, size, uppercase }) => (
  <StyledInput
    name={name}
    id={name}
    type={type}
    placeholder={placeholder}
    value={value}
    disabled={disabled}
    onChange={onChange}
    size={size}
    uppercase={uppercase}
  />
);

Input.propTypes = {
  /** Is input disabled */
  disabled: PropTypes.bool,
  /** Input placeholder text */
  placeholder: PropTypes.string,
  /** Input value */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** Function that changes input value */
  onChange: PropTypes.func.isRequired,
  /** input type */
  type: PropTypes.string,
  /** Input text uppercase */
  uppercase: PropTypes.bool,
  /** Input name */
  name: PropTypes.string.isRequired,
  /** Input id */
  id: PropTypes.string.isRequired,
  /** Input size */
  size: PropTypes.oneOf(["normal", "large"])
};

Input.defaultProps = {
  value: "",
  placeholder: "",
  disabled: false,
  uppercase: false,
  type: "text",
  size: "normal"
};

export default Input;
