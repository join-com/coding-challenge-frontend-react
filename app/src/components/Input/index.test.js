import React from 'react';
import toJson from 'enzyme-to-json';

import Input from './index';

/* global shallow, jest, mount */
describe('<Input />', () => {
    it('should render Input', () => {
        const wrapper = mount(<Input type="text" />);

        expect(wrapper.find('input')).toHaveLength(1);
    });

    it('should render Input type  but not :) text', () => {
        const wrapper = mount(<Input type="date" />);

        expect(wrapper.find('input').prop('type')).toBe('text');
    });

    it('should matches the snapshot', () => {
        const tree = mount(<Input />);

        expect(toJson(tree)).toMatchSnapshot();
    });
});
