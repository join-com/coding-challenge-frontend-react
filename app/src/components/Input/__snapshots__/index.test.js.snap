// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should matches the snapshot 1`] = `
<Input>
  <styled.input
    className="c-input"
  >
    <StyledComponent
      className="c-input"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "eMKLvb",
            "rules": Array [
              "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <input
        className="c-input sc-bdVaJa eMKLvb"
      />
    </StyledComponent>
  </styled.input>
</Input>
`;
